console/src/components/Services/Data/Common/ReusableComponents/RootFieldEditor.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import styles from '../../../../Common/Common.scss';
+import { getRootFieldLabel } from './utils';
+
+const RootFieldEditor = ({
+  aliases,
+  disabled,
+  selectOnChange,
+  selectByPkOnChange,
+  selectAggOnChange,
+  insertOnChange,
+  updateOnChange,
+  deleteOnChange,
+  tableName,
+}) => {
+  const {
+    select,
+    select_by_pk: selectByPk,
+    select_aggregate: selectAgg,
+    insert,
+    update,
+    delete: _delete,
+  } = aliases;
+
+  const [queryExpanded, setQueryExpanded] = React.useState(true);
+  const [mutationExpanded, setMutationExpanded] = React.useState(true);
+
+  const getDefaultRootField = rfType => {
+    if (rfType.includes('select')) {
+      return rfType.replace('select', tableName);
+    }
+    return `${rfType}_${tableName}`;
+  };
+
+  const getRow = (rfType, value, onChange) => (
+    <div
+      className={`${styles.display_flex} row ${styles.add_mar_bottom_small}`}
+    >
+      <div className={`${styles.add_mar_right} col-md-3`}>
+        {getRootFieldLabel(rfType)}
+      </div>
+      <div className={'col-md-5'}>
+        <input
+          type="text"
+          value={value || ''}
+          placeholder={`${getDefaultRootField(rfType)} (default)`}
+          className="form-control"
+          onChange={onChange}
+          disabled={disabled}
+        />
+      </div>
+    </div>
+  );
+
+  const getQuerySection = () => {
```

Comment:
Will it add value if we are able to refactor this functionality into a seperate function and consume it by args (for query and mutation)?

Reply:
Yeah. Reduces some repeated code. Changed.